internal/db/schema/migrations/oss/postgres/66/01_plugin_host_external_name.up.sql

```diff
@@ -0,0 +1,50 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: MPL-2.0
+
+begin;
+  alter table host_plugin_host
+    add column external_name text
+      -- As it currently stands, AWS restricts EC2 Instance names to 256 UTF-8
+      -- characters. Azure restricts their VM resource names to, at most, 64
+      -- characters, and also have a list of disallowed runes.
+      --
+      -- These constraints follow the lowest common denominator between both
+      -- providers for maximum compatibility. They are also synced with their
+      -- logical counterpart (see `NewHost` in internal/host/plugin/host.go) to
+      -- prevent a situation where a bad name could make an update job stop
+      -- working.
+      constraint external_name_only_has_printable_characters
+        check (external_name is null or external_name !~ '[^[:print:]]')
+      constraint external_name_has_max_256_characters
+        check (external_name is null or length(external_name) <= 256);
```

Comment:
* Are empty strings allowed?
* No need for the null check (see above)

Reply:
> Are empty strings allowed?

No, they are not. This new field is either `NULL` or an actual string (with len > 0).

> No need for the null check (see above)

TIL! Got it - Thanks for letting me know.

Reply:
> Should the constraint have a length(external_name) > 0 then?

Looks like that's a yes, resolving this thread in favour of https://github.com/hashicorp/boundary/pull/3074#discussion_r1153505016